src/components/SettingsBox.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import ToggleSwitch, { ToggleSwitchName } from "./ToggleSwitch";
+import styled from "styled-components";
+
+const ModuleTitle = styled.h3`
+  color: #fff;
+  font-weight: 700;
+  margin-top: 10px;
+  margin-bottom: 10px;
+`;
+
+const SettingsBoxWrapper = styled.div`
+  width: 175px;
+  height: fit-content;
+  padding: 0px 10px 10px 10px;
+  background-color: transparent;
+  border: solid 5px #fff;
+  border-radius: 15px;
+`;
+
+const ToggleSwitchWrapper = styled.div`
+  width: 100%;
+  height: 30px;
+  display: flex;
+  align-items: center;
+`;
+
+const SettingsBox = ({ settings, setSettings }) => {
```

Comment:
This is a mix of a controlled and uncontrolled component.
`settings` is the controlled part (parent controls the value).
`setSettings` is the uncontrolled part (parent has no control when and how the `SettingsBox` is using it).

Instead, you should bypass an event like `onChange` if you like to have it controlled.

Reply:
very good point, thank you. I finally got the controlled/uncontrolled thing figured out